### ISSUE TITLE
Update to PostCSS 7

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#3477]
We [did](https://github.com/stylelint/stylelint/pull/3469) updated to PostCSS 7. This update doesn't break functionality, but it introduces [a lot of warnings in tests](https://travis-ci.org/stylelint/stylelint/jobs/405145270) and while using stylelint CLI:

```
You did not set any plugins, parser, or stringifier. Right now PostCSS do nothing. Pick plugins for your case on https://www.postcss.parts/ and usem them in postcss.config.js.
```

Our users will see this warnings, if we wouldn't change anything in our code.

As we discussed with @ai in https://github.com/postcss/postcss/issues/1171, stylelint team needs to investigate where PostCSS is used “incorrectly” in stylelint codebase. It might be a correct usage after all, but we need to investigate.

Because we're going to release in few days, I'm going to roll back all PostCSS 7 related updates.

**Comment by ntwb** [comment]
Agreed, I just poked around a little and this will require significant work

Great work catching this @hudochenkov 💯
